### DISCUSSION TITLE
Use filtered data(s) to represent them in chart/graph

### DISCUSSION THREAD

**Discussion by MuhammadTanvir** [Power-Components/livewire-powergrid#641] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary

I would like to use filtered data(s) to represent them in chart/graph. How can I get the filtered requests and use them in query on the controller?
Any idea/ suggestion would be appreciated. Thank you for this wonderful pacakge

**Comment by akulmehta** [comment]
I think this is possible but I have not tried it. Here is my suggestion:
1. Publish the views using `php artisan vendor:publish --tag=livewire-powergrid-config`
2. Go to file `resources\views\vendor\livewire-powergrid\components\table.blade.php`. In this file the `$data` variable holds the data which is displayed after filtering/searching. You can dump it or put it as json in a hidden text area in the same file and use a JS library like D3js to pull it out.
3. You may need to add emit functions from the livewire component and listen in the JS for these events to update your charts based on the updates in filtering/searching.

I hope this helps.

**Reply by MuhammadTanvir** [reply]
thank you for your suggestion

**Reply by akulmehta** [reply]
Please let us know what works for you.

**Reply by MuhammadTanvir** [reply]
I have implemented the chart with ajax

**Reply by musumbidenis** [reply]
Hello @MuhammadTanvir,
Kindly show how you achieved that using ajax

**Reply by luanfreitasdev** [reply]
Curious to understand !

**Reply by MuhammadTanvir** [reply]
I have used the ajax onchange event to catch requests of the filter options and use them on the controller. By using necessary queries, I represent the data as a chart on the view file

**Reply by musumbidenis** [reply]
That's great. Kindly share some code to shade more light on how you used ajax onchange event to catch filter option request

**Reply by MuhammadTanvir** [reply]
Hi @musumbidenis, as it is a private project, I have no permission to share the codes, sorry for that. But I can give some hints...

```
$('#input_date_formatted')
 .on('change', function() {
     filter_data();
}); 

 function filter_data() {
 var input_date_formatted = $('#input_date_formatted').val();  
} 
```      
you can use the code above to catch the filter option request.

**Reply by musumbidenis** [reply]
Thank you so much.
